js/ui/map.js

```diff
@@ -518,6 +520,7 @@ var Map = Widget.inherit({
                 this._dimensionChanged();
                 break;
             case "provider":
+                this._suppressAsyncAction = true;
```

Comment:
I don't like an idea to add a flag. It is also quite interesting that there is one more clearing in the '_clean' method.